### PR TITLE
Re-write Fides security best practices guide

### PR DESCRIPTION
**Pull request by daveqnet**
### Description of Changes

* The Fides security best practices guide has been re-written from scratch following internal review.

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created

### CODE REVIEW

File: pages/fides/installation/security_practices.md [hunk 1]
```diff
@@ -1,48 +1,170 @@
...
+
+The Fides webserver allows all inbound traffic to all API endpoints and front-end paths. No traffic is denied. This includes potentially dangerous fides API endpoints such as `/api/v1/admin/db/reset`.
+
+Most Fides API endpoints must not be exposed to users on untrusted networks such as the public Internet. Therefore an ingress or API gateway must be placed in front of the Fides webserver to allow-list only the minimum set of routes required by external users.
+
+![Public Ingress Route Allow Listing](https://user-images.githubusercontent.com/83430497/219023171-6f950b7c-0307-4557-b285-ec744e25780c.png)
+
+#### Public Internet  --> Fides Webserver
+
+Only the following endpoints should be allow-listed from the public Internet to the Fides webserver.
```
Comment (ThomasLaPiana) [inline pages/fides/installation/security_practices.md:39]: Asking for knowledge, how was this list determined?

Comment (daveqnet) [reply pages/fides/installation/security_practices.md:39]: Hi @ThomasLaPiana, I generated this list by deploying Fides locally, manually crawling the Privacy Center UI, and capturing all the backend calls made by the browser along the way. Kind of a black box approach.

The only exception to the above would be the oauth callback endpoint, which I knew would need to be publicly accessible for SaaS connectors to work.

@NevilleS verified this list by inspecting the code, but I'm not sure exactly how he did that.

Comment (daveqnet) [reply pages/fides/installation/security_practices.md:39]: > ...manually crawling the Privacy Center UI, and capturing all the backend calls made by the browser along the way.

If you're curious, the tool I used was [Burp Suite](https://portswigger.net/burp). It's very useful for security testing.

File: pages/fides/installation/security_practices.md [hunk 2]
```diff
@@ -1,48 +1,170 @@
...
+
+### Public Ingress: TLS Termination of Inbound Traffic
+
+The Fides webserver and Privacy Center services are not designed for a deployment with direct termination of TLS for inbound traffic. These services must not be exposed to users on untrusted networks. Therefore a reverse proxy or load balancer must be placed in front of them to enforce encryption of inbound traffic using TLS. 
+
+![Public Ingress TLS Termination Diagram](https://user-images.githubusercontent.com/83430497/219023044-bb8552aa-cbc4-4842-bb91-2fd1d284cf82.png)
+
+### Public Ingress: Route Allow-Listing for Inbound Traffic 
+
+The Fides webserver allows all inbound traffic to all API endpoints and front-end paths. No traffic is denied. This includes potentially dangerous fides API endpoints such as `/api/v1/admin/db/reset`.
```
Comment (ThomasLaPiana) [inline pages/fides/installation/security_practices.md:31]: Does this change given that we now have the `end = "prod"` configuration value available?

**Comment by ThomasLaPiana** [comment]
@daveqnet heads up that I have a PR open here too: https://github.com/ethyca/fidesdocs/pull/57

Might be worth reading in case it affects anything in here

File: pages/fides/installation/security_practices.md (hunk 2 above)
Comment (daveqnet) [reply pages/fides/installation/security_practices.md:31]: You're right, I need to re-write this now that API auth is possible.

**Pull request merged**